tests/test_regression.py

```diff
@@ -110,3 +113,20 @@ def mocking_function(*args, **kwargs):
 
 def test_p4_failed_opened(perforce_environment, mock_opened):
     assert not __main__.run(perforce_environment.settings)
+
+
+# TODO: find a way to mock 'p4 opened' in Docker and move this test to 'test_api.py'
+def test_p4_api_failed_opened(perforce_environment, mock_opened):
+    step_name = "API"
+    config = f"""
+from universum.configuration_support import Configuration
+
+configs = Configuration([dict(name="{step_name}", artifacts="output.json",
+                              command=["bash", "-c", "{python()} -m universum api file-diff > output.json"])])
+    """
+    settings = shelve_config(config, perforce_environment)
+
+    assert not __main__.run(settings)
+    log = perforce_environment.temp_dir.join('artifacts', f'{step_name}_log.txt').read()
```

Comment:
Please create dedicated `artifact_dir` variable in TestEnvironment and use it here.